// @vx/network/esm/Graph.js
import _pt from "prop-types";
import React from 'react';
import { Group } from '@vx/group';
import Links from './Links';
import Nodes from './Nodes';
import DefaultNode from './DefaultNode';
import DefaultLink from './DefaultLink';
export default function Graph(_ref) {
  var graph = _ref.graph,
      _ref$linkComponent = _ref.linkComponent,
      linkComponent = _ref$linkComponent === void 0 ? DefaultLink : _ref$linkComponent,
      _ref$nodeComponent = _ref.nodeComponent,
      nodeComponent = _ref$nodeComponent === void 0 ? DefaultNode : _ref$nodeComponent,
      top = _ref.top,
      left = _ref.left;
  return graph ? React.createElement(Group, {
    top: top,
    left: left
  }, React.createElement(Links, {
    links: graph.links,
    linkComponent: linkComponent
  }), React.createElement(Nodes, {
    nodes: graph.nodes,
    nodeComponent: nodeComponent
  })) : null;
}
Graph.propTypes = {
  graph: _pt.any,
  linkComponent: _pt.oneOfType([_pt.string, _pt.any, _pt.elementType]),
  nodeComponent: _pt.oneOfType([_pt.string, _pt.any, _pt.elementType]),
  top: _pt.number,
  left: _pt.number
};
