// @vx/network/esm/Nodes.js
import _pt from "prop-types";
import React from 'react';
import cx from 'classnames';
import { Group } from '@vx/group';
import DefaultNode from './DefaultNode';
export default function Nodes(_ref) {
  var _ref$nodes = _ref.nodes,
      nodes = _ref$nodes === void 0 ? [] : _ref$nodes,
      _ref$nodeComponent = _ref.nodeComponent,
      nodeComponent = _ref$nodeComponent === void 0 ? DefaultNode : _ref$nodeComponent,
      className = _ref.className,
      _ref$x = _ref.x,
      x = _ref$x === void 0 ? function (d) {
    return d && d.x || 0;
  } : _ref$x,
      _ref$y = _ref.y,
      y = _ref$y === void 0 ? function (d) {
    return d && d.y || 0;
  } : _ref$y;
  return React.createElement(React.Fragment, null, nodes.map(function (node, i) {
    return React.createElement(Group, {
      key: "network-node-" + i,
      className: cx('vx-network-node', className),
      left: x(node),
      top: y(node)
    }, React.createElement(nodeComponent, {
      node: node
    }));
  }));
}
Nodes.propTypes = {
  nodes: _pt.arrayOf(_pt.any),
  nodeComponent: _pt.oneOfType([_pt.string, _pt.any, _pt.elementType]).isRequired,
  className: _pt.string,
  x: _pt.func,
  y: _pt.func
};
